File: Refactor.java
import java.io.File;
import java.nio.file.Files;
import java.nio.file.StandardOpenOption;
import java.util.Arrays;
import java.util.List;

public class Refactor {
    public static void main(String[] args) throws Exception {
        String baseDir = "backend/main/java/com/example/hrm";
        List<String> domains = Arrays.asList("users", "department", "employee", "tasks");
        List<String> layers = Arrays.asList("controller", "service", "repository", "dto", "mapper", "entity");

        for (String domain : domains) {
            File domainDir = new File(baseDir, domain);
            if (!domainDir.exists()) continue;

            for (String layer : layers) {
                new File(domainDir, layer).mkdirs();
            }

            File[] files = domainDir.listFiles();
            if (files == null) continue;

            for (File file : files) {
                if (file.isDirectory() || !file.getName().endsWith(".java")) continue;

                String name = file.getName();
                String layer = "entity";
                if (name.endsWith("Controller.java")) layer = "controller";
                else if (name.endsWith("Service.java")) layer = "service";
                else if (name.endsWith("Repository.java")) layer = "repository";
                else if (name.endsWith("DTO.java")) layer = "dto";
                else if (name.endsWith("Mapper.java")) layer = "mapper";

                File newFile = new File(domainDir, layer + "/" + name);
                
                String content = new String(Files.readAllBytes(file.toPath()));
                
                String oldPkg = "package com.example.hrm." + domain + ";";
                String newPkg = "package com.example.hrm." + domain + "." + layer + ";";
                content = content.replace(oldPkg, newPkg);
                
                String imports = "";
                if (domain.equals("users")) imports = "import com.example.hrm.users.dto.*;\nimport com.example.hrm.users.entity.*;\nimport com.example.hrm.employee.entity.*;\n";
                if (domain.equals("department")) imports = "import com.example.hrm.department.dto.*;\nimport com.example.hrm.department.entity.*;\nimport com.example.hrm.employee.entity.*;\n";
                if (domain.equals("employee")) imports = "import com.example.hrm.employee.dto.*;\nimport com.example.hrm.employee.entity.*;\nimport com.example.hrm.department.repository.*;\nimport com.example.hrm.users.repository.*;\nimport com.example.hrm.department.entity.*;\nimport com.example.hrm.users.entity.*;\nimport com.example.hrm.tasks.entity.*;\n";
                if (domain.equals("tasks")) imports = "import com.example.hrm.tasks.dto.*;\nimport com.example.hrm.tasks.entity.*;\nimport com.example.hrm.employee.entity.*;\n";

                content = content.replace(newPkg, newPkg + "\n\n" + imports);
                
                content = content.replace("import com.example.hrm.users.Role;", "import com.example.hrm.users.entity.Role;");
                content = content.replace("import com.example.hrm.users.User;", "import com.example.hrm.users.entity.User;");
                content = content.replace("import com.example.hrm.employee.Employee;", "import com.example.hrm.employee.entity.Employee;");
                content = content.replace("import com.example.hrm.department.Department;", "import com.example.hrm.department.entity.Department;");
                content = content.replace("import com.example.hrm.department.Position;", "import com.example.hrm.department.entity.Position;");
                content = content.replace("import com.example.hrm.tasks.Task;", "import com.example.hrm.tasks.entity.Task;");
                content = content.replace("import com.example.hrm.tasks.TaskStatus;", "import com.example.hrm.tasks.entity.TaskStatus;");
                content = content.replace("import com.example.hrm.employee.AttendanceStatus;", "import com.example.hrm.employee.entity.AttendanceStatus;");
                content = content.replace("import com.example.hrm.employee.LeaveStatus;", "import com.example.hrm.employee.entity.LeaveStatus;");
                content = content.replace("import com.example.hrm.users.UserRepository;", "import com.example.hrm.users.repository.UserRepository;");

                Files.write(newFile.toPath(), content.getBytes(), StandardOpenOption.CREATE, StandardOpenOption.TRUNCATE_EXISTING);
                file.delete();
            }
        }
        System.out.println("Refactoring complete.");
    }
}

File: refactor.py
import os

base_dir = r"c:\Users\ADMIN\OneDrive\Máy tính\hr-management\backend\main\java\com\example\hrm"
domains = ["users", "department", "employee", "tasks"]
layers = ["controller", "service", "repository", "dto", "mapper", "entity"]

import_additions = {
    "users": ["com.example.hrm.users.dto.*", "com.example.hrm.users.entity.*", "com.example.hrm.employee.entity.*"],
    "department": ["com.example.hrm.department.dto.*", "com.example.hrm.department.entity.*", "com.example.hrm.employee.entity.*"],
    "employee": ["com.example.hrm.employee.dto.*", "com.example.hrm.employee.entity.*", "com.example.hrm.department.repository.*", "com.example.hrm.users.repository.*", "com.example.hrm.department.entity.*", "com.example.hrm.users.entity.*", "com.example.hrm.tasks.entity.*"],
    "tasks": ["com.example.hrm.tasks.dto.*", "com.example.hrm.tasks.entity.*", "com.example.hrm.employee.entity.*"]
}

for domain in domains:
    domain_path = os.path.join(base_dir, domain)
    if not os.path.exists(domain_path): continue
    
    for layer in layers:
        os.makedirs(os.path.join(domain_path, layer), exist_ok=True)
        
    for file in os.listdir(domain_path):
        if not file.endswith(".java"): continue
        
        file_path = os.path.join(domain_path, file)
        
        # Determine layer
        if file.endswith("Controller.java"): layer = "controller"
        elif file.endswith("Service.java"): layer = "service"
        elif file.endswith("Repository.java"): layer = "repository"
        elif file.endswith("DTO.java"): layer = "dto"
        elif file.endswith("Mapper.java"): layer = "mapper"
        else: layer = "entity"
        
        new_path = os.path.join(domain_path, layer, file)
        
        with open(file_path, "r", encoding="utf-8") as f:
            content = f.read()
            
        old_pkg = f"package com.example.hrm.{domain};"
        new_pkg = f"package com.example.hrm.{domain}.{layer};"
        content = content.replace(old_pkg, new_pkg)
        
        imports = "\n".join([f"import {imp};" for imp in import_additions[domain]])
        content = content.replace(new_pkg, new_pkg + "\n\n" + imports)
        
        content = content.replace("import com.example.hrm.users.Role;", "import com.example.hrm.users.entity.Role;")
        content = content.replace("import com.example.hrm.users.User;", "import com.example.hrm.users.entity.User;")
        content = content.replace("import com.example.hrm.employee.Employee;", "import com.example.hrm.employee.entity.Employee;")
        content = content.replace("import com.example.hrm.department.Department;", "import com.example.hrm.department.entity.Department;")
        content = content.replace("import com.example.hrm.department.Position;", "import com.example.hrm.department.entity.Position;")
        content = content.replace("import com.example.hrm.tasks.Task;", "import com.example.hrm.tasks.entity.Task;")
        content = content.replace("import com.example.hrm.tasks.TaskStatus;", "import com.example.hrm.tasks.entity.TaskStatus;")
        content = content.replace("import com.example.hrm.employee.AttendanceStatus;", "import com.example.hrm.employee.entity.AttendanceStatus;")
        content = content.replace("import com.example.hrm.employee.LeaveStatus;", "import com.example.hrm.employee.entity.LeaveStatus;")
        content = content.replace("import com.example.hrm.users.UserRepository;", "import com.example.hrm.users.repository.UserRepository;")

        with open(new_path, "w", encoding="utf-8") as f:
            f.write(content)
            
        os.remove(file_path)

print("Refactoring complete.")
